src/Grpc.Net.Client/Internal/GrpcCall.cs

```diff
@@ -811,9 +811,15 @@ private void SetFailedResult(Status status)
         // 1. Diagnostic source is enabled
         // 2. Logging is enabled
         // 3. There is an existing activity (to enable activity propagation)
-        if (diagnosticSourceEnabled || Logger.IsEnabled(LogLevel.Critical) || Activity.Current != null)
+        // 4. ActivitySource has listeners
+        if (diagnosticSourceEnabled || Logger.IsEnabled(LogLevel.Critical) || Activity.Current != null || GrpcDiagnostics.ActivitySource.HasListeners())
         {
-            activity = new Activity(GrpcDiagnostics.ActivityName);
+            activity = GrpcDiagnostics.ActivitySource.CreateActivity(GrpcDiagnostics.ActivityName, ActivityKind.Client);
+
+            // ActivitySource only returns an activity if someone is listening.
+            // If we're at this point then we always want there to be an activity. Create the activity manually.
+            activity ??= new Activity(GrpcDiagnostics.ActivityName);
```

Comment:
The default is internal, but it can't be set when using the `Activity` ctor.

Reply:
We can consider in the future release to add a constructor that can take a `ActivityKind`. I think this can help in such scenarios but still will not help older Grpc client versions that have already been released.